search-app/src/cmr/search/services/query_execution/facets/hierarchical_v2_facets.clj

```diff
@@ -354,11 +378,25 @@
         (assoc updated-facet :children (:children earth-science-facets))))
     hierarchical-facet))
 
+(defn get-field-hierarchy
+  "Returns the field hierarchy for the given field and passed in query parameters."
+  [field query-params]
+  (let [field-hierarchy (nested-fields-mappings field)]
+    (if (not= :temporal-facet field)
+      field-hierarchy
+      (let [lowest-field (temporal-facets/query-params->time-interval query-params)
+            relevant-fields (concat
```

Comment:
This probably deserves its own function, which would clean up the `let` block and allow you to simply return the function call.

Reply:
Refactored, definitely better.